### PR TITLE
vala: update to 0.36.6 use gcc and fix tests

### PR DESCRIPTION
**Pull request by ghost**
Bug fixes and improved build

### CODE REVIEW

File: components/developer/vala/Makefile [hunk 1]
```diff
@@ -52,6 +56,10 @@ build:      $(BUILD_32_and_64)
 
 install:    $(INSTALL_32_and_64)
 
+# Failed tests: /methods/bug653391 - cc1: some warnings being treated as errors
```
Comment (pyhalov) [inline components/developer/vala/Makefile]: Better put test results in test/results.master, set COMPONENT_TEST_RESULTS and use COMPONENT_TEST_TRANSFORMS to make them reproducible.

File: components/developer/vala/Makefile [hunk 2]
```diff
@@ -44,6 +43,11 @@ include $(WS_MAKE_RULES)/ips.mk
 
 PATH=$(PATH.gnu)
 
+COMPONENT_PRE_CONFIGURE_ACTION = ($(CLONEY) $(SOURCE_DIR) $(@D))
+
+CONFIGURE_SCRIPT = $(@D)/configure
+
+CONFIGURE_OPTIONS  +=  CC=gcc
```
Comment (pyhalov) [inline components/developer/vala/Makefile]: don't set this option, CC=$(CC) is already passed to configure by default.

File: components/developer/vala/patches/tests.patch [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+@@ -29,7 +29,11 @@
+ export G_DEBUG=fatal_warnings
+ 
+ VALAC=$topbuilddir/compiler/valac$EXEEXT
+-VALAFLAGS="$VALAFLAGS --vapidir $vapidir --disable-warnings --main main --save-temps -X -g -X -O0 -X -pipe -X -lm -X -DGETTEXT_PACKAGE=\"valac\" -X -Werror=return-type -X -Werror=init-self -X -Werror=implicit -X -Werror=sequence-point -X -Werror=return-type -X -Werror=uninitialized -X -Werror=pointer-arith -X -Werror=int-to-pointer-cast -X -Werror=pointer-to-int-cast -X -Wformat -X -Werror=format-security -X -Werror=format-nonliteral -X -Werror=redundant-decls -X -Werror=int-conversion"
++#VALAFLAGS="$VALAFLAGS --vapidir $vapidir --disable-warnings --main main --save-temps -X -g -X -O0 -X -pipe -X -lm -X -DGETTEXT_PACKAGE=\"valac\" -X -Werror=return-type -X -Werror=init-self -X -Werror=implicit -X -Werror=sequence-point -X -Werror=return-type -X -Werror=uninitialized -X -Werror=pointer-arith -X -Werror=int-to-pointer-cast -X -Werror=pointer-to-int-cast -X -Wformat -X -Werror=format-security -X -Werror=format-nonliteral -X -Werror=redundant-decls -X -Werror=int-conversion"
++# Hardcoded --cc flag for now; Use ./configure CC=gcc when building vala
```
Comment (pyhalov) [inline components/developer/vala/patches/tests.patch]: Don't hardcode --cc=gcc, use --cc=$(CC) and set COMPONENT_TEST_ENV += CC=$(CC)

File: components/developer/vala/vala.p5m [hunk 4]
```diff
@@ -10,7 +10,7 @@
 #
 
 #
-# Copyright 2017 Alexander Pyhalov
+# Copyright 2017 Jason Martin
```
Comment (pyhalov) [inline components/developer/vala/vala.p5m]: This file didn't change. Better add your copyright to Makefile

File: components/developer/vala/patches/tests.patch [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
++	-X -Werror=init-self \
++	-X -Werror=implicit \
++	-X -Werror=sequence-point \
++	-X -Werror=return-type \
++	-X -Werror=uninitialized \
++	-X -Werror=pointer-arith \
++	-X -Wformat \
++	-X -Werror=format-security \
++	-X -Werror=format-nonliteral \
++	-X -Werror=redundant-decls"
```
Comment (pyhalov) [inline components/developer/vala/patches/tests.patch:42]: what cc will VALAC use?

**Comment by ghost** [comment]
I think it is picked up automatic now. I may be able to remove line in Makefile now,

**Comment by pyhalov** [comment]
Does it pick up /usr/gcc/4.9/bin/gcc or /usr/bin/gcc ?

**Comment by ghost** [comment]
/usr/gcc/4.9/bin/gcc is showing in tests Makefile right now

**Comment by ghost** [comment]
after removing line from Makefile

**Comment by ghost** [comment]
Sorry, I will have to put line back give me a few minutes.

**Pull request merged**